### PR TITLE
mgr/dashboard: Add Prettier formatter to the frontend

### PR DESCRIPTION
**Pull request by tspmelo**
This will force the use of prettier on frontend code,
bringing a more consistent formatting to all the code.

The current configuration will apply the prettier formatting,
on all staged files, during the precommit hook.

`Signed-off-by: Tiago Melo <tmelo@suse.com>`

### CODE REVIEW

**Comment by votdev** [comment]
Why do we allow [x, y, z] but not {x: 0, y: 1, z: 2}? This does not make sense to me.

File: src/pybind/mgr/dashboard/frontend/src/app/shared/validators/cd-validators.ts [hunk 1]
```diff
@@ -49,17 +40,20 @@ export class CdValidators {
...
+          return control.parent && control.parent.get(key).value === prerequisites[key];
+        })
+      ) {
         return null;
       }
-      const success = _.isFunction(condition) ? condition.call(condition, control.value) :
-        isEmptyInputValue(control.value);
-      return success ? {'required': true} : null;
+      const success = _.isFunction(condition)
+        ? condition.call(condition, control.value)
```
Comment (Devp00l) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/validators/cd-validators.ts]: Don't like the operator at the beginning of the line - it should be at the end of the last line to be more clear about that the line is continuing.

```
question ?
yes :
no
```

Comment (tspmelo) [reply src/pybind/mgr/dashboard/frontend/src/app/shared/validators/cd-validators.ts]: ATM you can't change how this is formatted. 
There is an open discussion over at prettier repo regarding this: https://github.com/prettier/prettier/issues/737 .

I don't mind this, with this you can look at 3 consecutive long lines and figure out what each does.

Regarding the end of the line, it's the same situation as when you have multiple piped methods, you know its continuing when there is no `;` on the end of the line and the next method is on a new line.

**Comment by tspmelo** [comment]
@votdev 
> Why do we allow [x, y, z] but not {x: 0, y: 1, z: 2}? This does not make sense to me.

We already had a rule to force a space in the beginning and end of `{ }`, so it makes sense to keep it that way.
In regards to the list, we can't change that format rule in prettier.

**Comment by ricardoasmarques** [comment]
> In regards to the list, we can't change that format rule in prettier.
In that case, I wonder if we should remove the white spaces on objects for consistency with arrays.

**Comment by tspmelo** [comment]
@ricardoasmarques @votdev I kind of like it the way it is ATM. 
Most of the places I have checked use it this way, p.e.: [MDN Array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array) and the TS import declaration usually have spaces after the brackets.

But if that is the deciding factor I can change the `Bracket Spacing` to false.

**Comment by votdev** [comment]
I've checked https://developer.mozilla.org/enUS/docs/Web/JavaScript/Reference/Global_Objects/Object/keys and they use the syntax for arrays and objects proposed by this PR, so i'm fine with that. But i still do not understand why there is a difference between arrays and objects.

File: src/pybind/mgr/dashboard/frontend/src/app/shared/validators/cd-validators.ts (hunk 1 above)
Comment (Devp00l) [reply src/pybind/mgr/dashboard/frontend/src/app/shared/validators/cd-validators.ts]: The issue mentioned is not the right issue - I mend prettier/prettier#716 ,
but reading this I'm OK with it.

**Pull request merged**